mmhuman3d/core/cameras/camera_parameters.py

```diff
@@ -483,12 +487,45 @@ def __check_value_type__(self, key: Any, val: Any) -> None:
         Raises:
             TypeError:
                 Value is supported but doesn't match definition.
+
+        Returns:
+            val (Any): The value casted into correct format.
         """
-        if type(val) != self.__class__.SUPPORTED_KEYS[key]['type']:
+        np_type_mapping = {int: np.integer, float: np.floating}
+        supported_keys = self.__class__.SUPPORTED_KEYS
+        check_passed = False
+        ret_val = None
+        if supported_keys[key]['type'] == int or\
+                supported_keys[key]['type'] == float:
+            type_str = str(type(val))
+            if type(val) == self.__class__.SUPPORTED_KEYS[key]['type']:
+                check_passed = True
+                ret_val = val
+            elif 'numpy' in type_str:
+                # a value is required, not array
+                if np.issubdtype(
```

Comment:
The `item()` method cannot distinguish between an array and a value, but `type()` does. To prevent user passes an array in shape like (1, 1), `type()` is essential.
```python
>>> import numpy as np
>>> arr = np.zeros(3, dtype=np.int64)
>>> arr[0:1].item()
0
>>> arr[0].item()
0
```
For pytorch, no method has been proved effective as above, all Tensor are treated as array, thus we use `item()`.

Reply:
As discussed offline, we may add a warning message.

Reply:
For torch.Tensor, length of size is used to pick value from array:
```python
>>> tensor = torch.zeros(size=(3,), dtype=torch.int64)
>>> type(tensor[0:1])
<class 'torch.Tensor'>
>>> type(tensor[0])
<class 'torch.Tensor'>
>>> tensor[0:1].size()
torch.Size([1])
>>> tensor[0].size()
torch.Size([])
```
Error message for array input will be added soon.

Reply:
When an array is found, the error message will be different (a bit longer) than normal case. See details in 4056412.